src/bp-members/classes/class-bp-members-notices-rest-controller.php

```diff
@@ -0,0 +1,1090 @@
+						'context'     => array( 'view', 'edit', 'embed' ),
+						'description' => __( 'Content of the community notice.', 'buddypress' ),
+						'type'        => 'object',
+						'required'    => true,
+						'arg_options' => array(
+							'sanitize_callback' => null,
+							'validate_callback' => null,
+						),
+						'properties'  => array(
+							'raw'      => array(
+								'description' => __( 'Content for the community notice, as it exists in the database.', 'buddypress' ),
+								'type'        => 'string',
+								'context'     => array( 'edit' ),
+							),
+							'rendered' => array(
+								'description' => __( 'HTML content for the community notice, transformed for display.', 'buddypress' ),
+								'type'        => 'string',
+								'context'     => array( 'view', 'edit', 'embed' ),
+								'readonly'    => true,
+							),
+						),
+					),
+					'target'      => array(
+						'context'     => array( 'view', 'edit', 'embed' ),
+						'description' => __( 'The target of the community notice.', 'buddypress' ),
+						'enum'        => array( 'admins', 'contributors', 'community' ),
+						'default'     => 'community',
+						'type'        => 'string',
+					),
+					'date'        => array(
+						'context'     => array( 'view', 'edit', 'embed' ),
+						'description' => __( 'The date of the community notice, in the site\'s timezone.', 'buddypress' ),
+						'readonly'    => true,
+						'type'        => array( 'string', 'null' ),
+						'format'      => 'date-time',
+					),
+					'date_gmt'    => array(
+						'context'     => array( 'view', 'edit', 'embed' ),
+						'description' => __( 'The date of the community notice, as GMT.', 'buddypress' ),
+						'readonly'    => true,
+						'type'        => array( 'string', 'null' ),
+						'format'      => 'date-time',
+					),
+					'priority'    => array(
+						'context'     => array( 'view', 'edit', 'embed' ),
+						'description' => __( 'The notice priority.', 'buddypress' ),
+						'readonly'    => true,
+						'type'        => 'integer',
+					),
+					'action_url'  => array(
+						'context'     => array( 'view', 'edit', 'embed' ),
+						'description' => __( 'The Notice Action URL.', 'buddypress' ),
+						'format'      => 'uri',
+						'type'        => 'string',
+					),
+					'action_text' => array(
+						'context'     => array( 'view', 'edit', 'embed' ),
+						'description' => __( 'The Notice Action text.', 'buddypress' ),
+						'type'        => 'string',
+					),
+				),
+			);
+		}
+
+		/**
+		 * Filters the notice schema.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param array $schema The endpoint schema.
+		 */
+		return apply_filters( 'bp_members_notices_rest_schema', $this->add_additional_fields_schema( $this->schema ) );
+	}
+
+	/**
+	 * Get the query params for notices collection.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @return array
+	 */
+	public function get_collection_params() {
+		$params                       = parent::get_collection_params();
```

Comment:
This is adding some fields which I'm not sure are supported currently. Like `search`.

Reply:
Good catch, I'll clean up these fields.

Reply:
Only the `search` field needed to be removed ✅